src/main/java/uk/gov/ida/verifyserviceprovider/services/ResponseService.java

```diff
@@ -72,16 +70,30 @@ public TranslatedResponseBody convertTranslatedResponseBody(
                 return responderCodeTranslator.translateResponderCode(statusCode);
             case StatusCode.SUCCESS:
                 List<Assertion> assertions = assertionDecrypter.decryptAssertions(validatedResponse);
+                if (assertionsContainEidasUnsignedAssertionsResponse(assertions)) {
+                    UnsignedAssertionsResponseHandler handler = unsignedAssertionsResponseHandler.get();
```

Comment:
_In no way a blocker, and probably just a style choice..._ The `unsignedAssertionsResponseHandler` is `Optional`, but we straight-up `get()` it. It feels like maybe it's not optional and always present? (Or always present when we need it?)

I guess so long as we throw an exception if it's not there when we need it, then that makes sense - but we can achieve that without `Optional`.

Reply:
Yeah, it does feel a bit weird. It's an Optional as this class can be instantiated in a couple of ways, depending on how the config is set up (eIDAS enabled or not). The conditional on line 73 means we're only ever running this line of code if we're in an eIDAS enabled VSP, so the `get()` _should_ be ok. But you're right, I should probably throw an Exception if it's not present. Will add it.